t2v_metrics/models/vqascore_models/gpt4v_model.py

```diff
@@ -44,97 +59,80 @@ def __init__(self,
+        for path in paths:
+            if path.lower().endswith(('.mp4', '.avi', '.mov', '.mkv')):  # Video file
+                frames = extract_frames(path, num_frames)
+                loaded_data.append({
+                    'path': path,
+                    'type': 'video',
+                    'frames': frames
+                })
+            else:  # Image file
+                loaded_data.append({
+                    'path': path,
+                    'type': get_image_type(path),
+                    'base64': encode_image(path)
+                })
+        return loaded_data
+
+    def forward_single(self, data, question, answer):
         try:
+            if data['type'] == 'video':
+                content = [
+                    {"type": "text", "text": question},
+                    *[{"type": "image_url", "image_url": {"url": f"data:image/jpg;base64,{frame}"}} for frame in data['frames']]
+                ]
+            else:
+                content = [
+                    {"type": "text", "text": question},
+                    {"type": "image_url", "image_url": {"url": f"data:image/{data['type']};base64,{data['base64']}"}}
+                ]
+
             completion = self.client.chat.completions.create(
                 model=self.model_name,
-                messages= [
-                {
-                    "role": "user",
-                    "content": [
-                    {
-                        "type": "text",
-                        "text": question
-                    },
-                    {
-                        "type": "image_url",
-                        "image_url": {
-                        "url": f"data:image/{image['type']};base64,{image['base64']}"
-                        }
-                    }
-                    ]
-                }
-                ],
+                messages=[{"role": "user", "content": content}],
                 logprobs=True,
                 top_logprobs=self.top_logprobs,
-                # logit_bias={yes_token:50, no_token:50}
             )
-        except:
-            print(f"Warning: completion not generated for image: {image['path']} and question: {question} and answer: {answer}")
-            print(f"Trying again with the same image")
-            try:
-                completion = self.client.chat.completions.create(model=self.model_name, messages= [{"role": "user","content": [{"type": "text", "text": question},{ "type": "image_url","image_url": {"url": f"data:image/{image['type']};base64,{image['base64']}"}}]}],logprobs=True,top_logprobs=self.top_logprobs,)
-            except:
-                print(f"Failed image: {image['path']} and question: {question} and answer: {answer}")
-                return torch.Tensor([0.0])
-
-        # print(completion.choices[0].message)
-        # print(completion.choices[0].logprobs)
-        # print(completion.choices[0].logprobs.content[0])
-        is_generated = False
+        except Exception as e:
```

Comment:
The reason why I have it tried again was because sometimes ChatGPT just glitched -- but the second time you try it will work

Reply:
Fixed